metpy/calc/cross_sections.py

```diff
@@ -0,0 +1,307 @@
+
+Compared to the rest of the calculations which are based around pint quantities, this module
+is based around xarray DataArrays.
+"""
+
+import cartopy.crs as ccrs
+import numpy as np
+import xarray as xr
+
+from .basic import coriolis_parameter
+from .tools import first_derivative
+from ..package_tools import Exporter
+from ..xarray import CFConventionHandler, check_matching_coordinates
+
+exporter = Exporter(globals())
+
+
+def distances_from_cross_section(cross):
+    """Calculate the distances in the x and y directions along a cross-section.
+
+    Parameters
+    ----------
+    cross : `xarray.DataArray`
+        The input DataArray of a cross-section from which to obtain geometeric distances in
+        the x and y directions.
+
+    Returns
+    -------
+    x, y : tuple of `xarray.DataArray`
+        A tuple of the x and y distances as DataArrays
+
+    """
+    if (CFConventionHandler.check_axis(cross.metpy.x, 'lon') and
+            CFConventionHandler.check_axis(cross.metpy.y, 'lat')):
+        # Use pyproj to obtain x and y distances
+        from pyproj import Geod
+
+        g = Geod(cross.metpy.cartopy_crs.proj4_init)
+        lon = cross.metpy.x
+        lat = cross.metpy.y
+
+        forward_az, _, distance = g.inv(lon[0].values * np.ones_like(lon),
+                                        lat[0].values * np.ones_like(lat),
+                                        lon.values,
+                                        lat.values)
+        x = distance * np.cos(np.deg2rad(90. - forward_az))
+        y = distance * np.sin(np.deg2rad(90. - forward_az))
+
+        # Build into DataArrays
+        x = xr.DataArray(x, coords=lon.coords, dims=lon.dims, attrs={'units': 'meters'})
+        y = xr.DataArray(y, coords=lat.coords, dims=lat.dims, attrs={'units': 'meters'})
+
+    elif (CFConventionHandler.check_axis(cross.metpy.x, 'x') and
+            CFConventionHandler.check_axis(cross.metpy.y, 'y')):
+
+        # Simply return what we have
+        x = cross.metpy.x
+        y = cross.metpy.y
+
+    else:
+        raise AttributeError('Sufficient horizontal coordinates not defined.')
+
+    return x, y
+
+
+def latitude_from_cross_section(cross):
+    """Calculate the latitude of points in a cross-section.
+
+    Parameters
+    ----------
+    cross : `xarray.DataArray`
+        The input DataArray of a cross-section from which to obtain latitudes.
+
+    Returns
+    -------
+    latitude : `xarray.DataArray`
+        Latitude of points
+
+    """
+    y = cross.metpy.y
+    if CFConventionHandler.check_axis(y, 'lat'):
+        return y
+    else:
+        latitude = ccrs.Geodetic().transform_points(cross.metpy.cartopy_crs,
+                                                    cross.metpy.x.values,
+                                                    y.values)[..., 1]
+        latitude = xr.DataArray(latitude, coords=y.coords, dims=y.dims,
+                                attrs={'units': 'degress_north'})
```

Comment:
What about southern latitudes? Not sure what the CF spec says here.

Reply:
In southern latitudes, it is negative. Here is what CF says about it: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#latitude-coordinate

(Also, I realized I misspelled `degrees_north`...oops.)